### ISSUE TITLE
Ability to remove a follower

### ISSUE THREAD

**Issue by aneillans** [bluesky-social/social-app#1160]
Feature request: Ability to remove a follower from your followers, either via soft block (block and unblock) or via a simple remove process.

At present if you block someone, and then unblock them later they are still following you .. this can be unexpected and is a behaviour pattern that is dissimilar to all other social networks. Additionally, there is no other way to remove a follower from your "herd".

**Comment by mschwendener** [comment]
Hi

Just out of curiosity: why would you want to remove a follower but not block them?

**Comment by aneillans** [comment]
> Hi
> 
> Just out of curiosity: why would you want to remove a follower but not block them?

Good question; its more a friend dynamic I've observed on other network occurring at times (not sure it's one I entirely subscribe to myself, but hey ho).  They seem to remove people they are ... not talking to ... but are in the same friend circles without outright blocking as it can cause issues on some networks when following conversation threads etc. 

However, I do fine the lack of automatic removal of follower status upon block unexpected (and unusual) behaviour.

**Comment by mschwendener** [comment]
> I do find the lack of automatic removal of follower status upon block unexpected (and unusual) behaviour.

I agree. 

I have another use case, not identical, but related: 

I'd like to be able to prevent bot accounts from following me with a setting in my preferences, so I don't have to block each of them manually. 

Recent examples (will add links in a moment):
poetic photos
90ies anime

and any other account that follows an absurdly high number of accounts. 

Which is not the same as an account that has a huge number of followers - that can happen and still be genuine.

**Comment by mschwendener** [comment]
> block bot accounts through a setting in my preferences

If it technically isn't easy to distinguish these accounts, I'd also be happy with a number setting, in analogy to "reply must have _n_ likes before I want to see it". 

For myself, I'd probably start at accounts which are following more than 5000 people, and lower the number later if necessary.

**Comment by mschwendener** [comment]
I'll start a separate issue for 2) and will come back here to clean this one up.

**Comment by aneillans** [comment]
> This is something we've discussed pretty heavily internally -- we don't yet know how to do it reliably.
> 
> The issue is that every user controls their own dataset. When I "follow" somebody, what I'm doing is publishing a follow record on my account. When I block somebody, same thing: I publish a block record. A block record can override the _behaviors_ of a follow record, but it can't force-remove the follow record from somebody else's repository.
> 
> We have discussed a voluntary removal of the follow record. That is, if somebody blocks me, my server automatically deletes my follow record. The only concern is that it's impossible to force. A server that's misbehaving (intentionally or not) wouldn't delete the follow record, so there would be some confusing scenarios where soft-blocks don't work. That led to some discussion of whether we could try to enforce voluntary behaviors by social pressure (eg "Follow the protocol or your server will be penalized") but we haven't established a framework for that kind of governance yet so we're hesitant to move forward with it.
> 
> A subsequent proposal to require "follow requests" has been discussed ([bluesky-social/atproto#1352](https://github.com/bluesky-social/atproto/issues/1352)) which is technically feasible but adds a little more friction than we want for the default model.
> 
> That leaves us where we currently are, which is tracking these two options but not ready to move on either yet.

Either the voluntary removal, or an additional record type of a literal soft-block to let you track or flag when a remote account should do that removal next time it refreshes or "sees" the records? Just trying to think about how it all works on the Fediverse, and while the same misbehaving server issue remains there (you will never remove that chance IMO in a distributed environment), but as you can say you can potentially govern on some avenues.

**Comment by littlejackal** [comment]
> Hi
> 
> Just out of curiosity: why would you want to remove a follower but not block them?

I agree with the original request although I can't exactly articulate why this is desired vs outright blocking without citing vague concepts like how blocklists are perceived and unnecessary follower association "feels".

Not being a public or notable figure it was plainly obvious which accounts were simply engaging in bot-like behaviour or follower-spam and which were genuinely interested in interacting with me and my content. I routinely pruned mass-follow accounts when it made sense, but it was by no means a scientific process.

From a day to day perspective there's no difference at all to my bsky experience whether "@buy-bitcoin-cheap-today.bsky.social" follows me or not; they're unlikely to ever actually interact with me in any meaningful way. But when I look at my follower list I see "friend, friend, friend, acquaintance, artist I'm interested in, news org, **obvious mass-follow account**, friend" and I'm just left with a vague desire to prune. On birdsite it was easy enough to block/unblock, took two seconds and I'd most likely never see the account again, but it sounds like there are technical reasons this may not be feasible here. 

Leaving a profile on a blocklist I found to have a connotation of engaging in (actively) unacceptable activity and being a profile I really and truly do not wish to interact or be associated with, but again, I can't cite anything other than personal preference on this. If blocking is the best feasible solution to a mass follow profile showing up on my "Followers" list then this is certainly not a hill I'm willing to die on -- I'll be happy to keep blocking mass follow accounts and lump them in with accounts I actively want to prevent interacting with me.

No real solutioning in this reply, but hopefully this mirror the way others feel.

**Comment by mackuba** [comment]
Yup, pretty much this @littlejackal. As someone with a fairly small account, I like to look at my follower lists / follower count, and see that I'm followed by let's say 200 people who are interested in what I post. Accounts like that anime pictures one that follows 100k profiles obviously aren't reading what I post, so they are just inflating that follower count artificially.

This obviously doesn't apply above some level of follower count where looking at the list or even reading follow notifications stops making sense.

**Comment by mschwendener** [comment]
I added a note to https://github.com/bluesky-social/social-app/issues/1161 that "can't become my follower" would be an option for such accounts, instead of automatically blocking them.

**Comment by TrechNex** [comment]
Another scenario might be users who are targeted online abuse or pile-ons. 

When you open Bluesky to the wider public, you may start to see people being "brigaded" by troll followers that organise themselves on certain online communities for any number of reasons. As things stand atm a user can block a user from interacting, but as stated above they will still show up in their "followers" list.

This could create a scenario where such communities can exploit that oversight and use it as a scorecard for accounts they've brigaded. Individual trolls could also use a script to check their own following list for users that have blocked them (because that is also publicly advertised information), and use that as a scorecard for "libs they've pwned".

The user who's been targeted would be left with a reminder of a nasty situation they found themselves in every time they check their own "followers" list. In addition, everyone would be able to see that they have been targeted in that way by looking at that user's "followers" list. Both of those things would be particularly not great if they were traumatised by the intensity of that harassment and want to move on with their lives without deleting/recreating their account.

My thinking as an outsider is that, even if the user can't remove a follower at the protocol level, the UI should hide followers that have been blocked by the user, and the user should have the option to hide/filter out followers that they have blocked. That way such information is not advertised anywhere other than "rogue instances" that most people won't join, and users who've been targeted feel that they have some agency in that situation. It's not a great fix, but it at least heads off that exploit before it becomes a problem.

**Comment by JadeNishikawa** [comment]
> Feature request: Ability to remove a follower from your followers, either via soft block (block and unblock) or via a simple remove process.
> 
> At present if you block someone, and then unblock them later they are still following you .. this can be unexpected and is a behaviour pattern that is dissimilar to all other social networks. Additionally, there is no other way to remove a follower from your "herd".

as of right now bluesky now has a bot issue because of people posting their invite codes on twitter/x right now you can block followers you don't want accessing your profile but it doesn't completely remove them i think follower removing like twitter/x has would be a great idea so you can keep tabs on real followers and weed out the fake ones

**Comment by serenacloos** [comment]
+1 Please add the option that let’s you unfollow.

**Comment by mackuba** [comment]
Copying here what I wrote on Bluesky about how this could be handled:

We could have some additional record type for this purpose, let's say an "unfollow" or "softblock" or "denyFollow" record. This would be like a reverse, negative follow, a record that you'd add to *your* repo, pointing in the opposite direction, indicating that you don't want to be followed by that account (it would be a permanent effect until you delete the record, not a one-time thing like classic "soft blocking" is on Twitter; the user would see the follow button disabled on your profile).

The way it would work is that AppView would take that into account and would make it *seem* as if the user is not following you, even though they technically have a follow record pointing to you. It would effectively make that follow from them to you disappear, even though the record is there.

The precedent for this is how posts are handled in AppView in the context of blocks and thread gates. If someone replies to you and you block them, that comment "disappears", even though there is still a post record in the author's repo - but AppView pretends that it doesn't exist. In the same way, it would pretend that the follow record doesn't exist.

**Comment by mackuba** [comment]
Alternatively, we could reuse the `block` record type and add multiple types of blocks, specified by some additional "type" field: you could have e.g. a block that blocks access to posts like it does now, a block that disallows any interaction but doesn't hide your posts, and a block that just disallows following like described above. (But this might be too hard to implement in the UI in a way that's easy to understand…)

**Comment by pfrazee** [comment]
> We could have some additional record type for this purpose, let's say an "unfollow" or "softblock" or "denyFollow" record. 

That really isn't an absurd thought, now that I consider it. It's still somewhat public but, you know, wcyd.

It's actually possible to do this permanently (disabling the follow option for the subject) or on a one-time-basis by simply including the URI of the follow you want to disable.

**Comment by rszasz** [comment]
Just a follow-up if this has been implemented or is still an open request.

**Comment by mackuba** [comment]
Nope, still just an idea at this point

**Comment by IPv6** [comment]
> Alternatively, we could reuse the `block` record type and add multiple types of blocks

Sounds reasonable and useful. There is a gradient of possible/wanted "feedback from other followers", where block is one of the extreme final "do not welcome here" status, and softblock more like "don`t bother me on each post please"

**Comment by ygg2** [comment]
I'm inclined to think both are useful - the idea of a public denyFollow record that prevents someone from ever following again seems like a nice feature, but it'd still make sense to be able to ask the server to remove its follow record, since then any other behavior that relies on follows would work as expected, and people wouldn't be able to look up who has softblocked them later.

A server could decide not to comply, but in the grand scheme of things, they can already be non-compliant in other areas like deleting content or removing spammers. Maybe request the server to delete the follow, but in the event a server doesn't comply, there can be an AppView override...?

**Comment by AdamTheoDotCom** [comment]
I really like this solution since it uses positive public records of approvals instead of a punitive "block-like" record. 

https://github.com/bluesky-social/atproto/issues/1352

**Comment by dvxbx** [comment]
I'm not a programmer or anything, but is it possible to separate the "follow" record which is just general from a "follower" record which is specific to me/belongs to my data?  Such a hypothetical "remove this follower" feature would mean that <i> as far as I'm concerned </i> the person doesn't follow me, they don't show up in my follower list (nor do I have to block them and create a record that you can look up on clearsky for example) although if you go to that person or bots' page I'm still listed as someone they follow. Unless they actually interact with me, it isn't necessary to block them, deny a follow, etc. or rely any other record other than that if you check <i>my</i> follower list, they just don't appear. And that "follower" record is basically just everyone that has followed me (which is public) minus my remove this follower list (which isn't public), or like, "check the public list against my list when you are displaying my followers". Does that make any sense?

**Comment by IPv6** [comment]
> I really like this solution since it uses positive public records of approvals instead of a punitive "block-like" record.

5 centes: facebook-like approves is a different approach to twitter's "block/softblock". And softblock imho better one. Approve puts a difficulties/discomfort on the follower, make it harder to subscribe. it is visible on facebook - friend requests are sent either by spammers or after private messaging. Denies hurts a little and after several denies users just dump the feauter.
On twitter following is much more lightweight and softblock is not a "deny" - it`s an option to not see follower in own feed, allowing him to see what he want in his/her seed. No regrets, no dumping feature :)

**Comment by scottdotweb** [comment]
> I really like this solution since it uses positive public records of approvals instead of a punitive "block-like" record.
> 
> [bluesky-social/atproto#1352](https://github.com/bluesky-social/atproto/issues/1352)

Blocking an account isn't "punitive".

> I really like this solution since it uses positive public records of approvals instead of a punitive "block-like" record.
> 
> [bluesky-social/atproto#1352](https://github.com/bluesky-social/atproto/issues/1352)

Blocking an account isn't "punitive" and it's an entirely erroneous direction to attempt to ascribe moral values to the data.

**Comment by valzargaming** [comment]
If this feature were implemented, wouldn't it also be beneficial to extend the ability to restrict access across the entire platform, rather than just individual users? For example, it could be useful to allow accounts to restrict access by permitting interactions only from followers associated with a specific @domain.tld, or conversely, to block all interactions from accounts not associated with a specific @domain.tld. By separating permissions into separate allow/disallow elements it would enable servers to enforce their own user's settings without infringing on the settings of others.

**Comment by chk-L** [comment]
So, a year later, there is no option to remove a follower? I'm new to this social media, so I'm not aware of everything, but I tried to block an account and it wasn't effective. Any alternatives?

**Comment by qtlunya** [comment]
Blocking an account will sort of make them effectively no longer follow you, but if you unblock them they will be following you again. Still no way around this as far as I know.

**Comment by mackuba** [comment]
I think Paul said he was working on some ideas for this last week

**Comment by surfdude29** [comment]
> I think Paul said he was working on some ideas for this last week

He did:

> Also fwiw some of these specific asks like list removal or soft blocks, we are also looking at how to get them done. I just wrote another protocol proposal for soft blocks. It bugs me that it’s hard

https://bsky.app/profile/pfrazee.com/post/3kzutz7zq3o2e

**Comment by perigozin** [comment]
Any updates on this? It's a critical feature in my opinion if Bluesky wants to be taken seriously.

**Comment by tengille** [comment]
Joined BlueSky and immediately got a "adult content" bot follower. I would like the ability to remove them either via block / unblock functionality OR a button that removes that follower when I go to their profile. 

When I block, they're _still_ following me. I don't want adult content bots to follow me.

**Comment by ICEknigh7** [comment]
> Just out of curiosity: why would you want to remove a follower but not block them?

To avoid a hostile reaction.

Blocking and then unblocking works wonders for this, at "the other place".

**Comment by dvxbx** [comment]
Its not just that, it is that there is a permanent record on ClearSky that you took the trouble to block someone. I know there are a lot of people who feel so great about blocking people for the slightest transgression, but I don't want any record whatsoever that some spambot ever interacted with my account, and especially if I would ever want to use my bluesky account for any professional capacity, especially in education, etc. "Just block them" is not a viable solution for me when they still technically follow me.

**Comment by Sininini** [comment]
I also hate how my followers number is inflated by bots and engagement farmers despite having blocked them.

**Comment by Simplyjennifer** [comment]
I find it incredibly surprising that blocking someone doesn’t remove them as a follow - It’s something I believe most people would find to be expected behavior on a social media platform. 

I am completely and utterly exhausted at the thought of unblocking every single block, unfollowing, and reblocking.

There’s no recourse to being added to a blocklist, which will often happen as people block all of a bad actor’s followers as a preemptive measure. 

I really want to love this app but I feel like I just lost my brand new toy.

**Comment by ajhih28** [comment]
I've tried to block and unblock someone, and the account still in my follower list.  That's frustrating. 
It's doesn't meant to unsee any of strangers,  I just want to let them not keep watching me. You know, some people or bot just random following some accounts and they don't care about them, but I do care.
 
I think it will be better if we can unfollow each other, when we want to know each other someday, in a normal way, then we have a chance to follow again.  But when it comes to block anyone, it sounds like I give no chance of it.

**Comment by ICEknigh7** [comment]
And there's also this related problem here: https://github.com/bluesky-social/social-app/issues/2028

It's impossible to unfollow somebody who blocked you.

**Comment by JoshMLeslie** [comment]
> Alternatively, we could reuse the `block` record type and add multiple types of blocks, specified by some additional "type" field: you could have e.g. a block that blocks access to posts like it does now, a block that disallows any interaction but doesn't hide your posts, and a block that just disallows following like described above. (But this might be too hard to implement in the UI in a way that's easy to understand…)

A bit aside, but while digesting this thread I whipped up a quick button as I imagined this. I think giving people granular control has merit.
![image](https://github.com/user-attachments/assets/4bde6a49-bbaf-4aa7-ad5c-7cb7f744c2fd)

**Comment by scottdotweb** [comment]
"A block that disallows any interaction but doesn't hide your posts" - I believe the name for this is "the Elon".

**Comment by Tyri0n** [comment]
Please add this. I don't wany any followers.

**Comment by surfdude29** [comment]
> And there's also this related problem here: #2028
> 
> It's impossible to unfollow somebody who blocked you.

It's impossible through the app, and I agree that should be fixed, but you can use this great tool to do it:

https://cleanfollow-bsky.pages.dev/

**Comment by cooljeanius** [comment]
> A bit aside, but while digesting this thread I whipped up a quick button as I imagined this. I think giving people granular control has merit. ![image](https://private-user-images.githubusercontent.com/30437767/385362038-4bde6a49-bbaf-4aa7-ad5c-7cb7f744c2fd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE3OTkyMTUsIm5iZiI6MTczMTc5ODkxNSwicGF0aCI6Ii8zMDQzNzc2Ny8zODUzNjIwMzgtNGJkZTZhNDktYmJhZi00YWE3LWFkNWMtN2NiN2Y3NDRjMmZkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTE2VDIzMTUxNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM4YTkwNGQ5ZDg3NjZlNWQyNWIyYWFlOTM2NTMzYzU1N2EzZjhmZmUyNTlmZDExNzBlOTVkMGM0NTUyNjYzNDcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.c76tq5p82Tv7vP8yLhJKLXwDZWqqFb1xdCz0fMt8LLM)
> 
> edit: comment attribution
> @JoshMLeslie 

I think something like this would be useful, but it'd be important to clarify the terminology first. Is "Block User" just the combination of the other two options, or would it do additional things?

**Comment by markski1** [comment]
Even if for some reason implementhis this explicitly was something "to be discussed". Is there any technical reason why blocking and unblocking doesn't force an unfollow? If it's just a matter of implementing it, is there any reason why it shouldn't be this way?

**Comment by ICEknigh7** [comment]
> * It's impossible to do this since there are users who have servers that are not run by Bluesky.
> * When you follow someone or block someone, it creates a record.
> * The block record basically overrides the behaviour of the follow record, but can't delete the follow record itself.
> * There can (and will) be servers that will simply disrespect or ignore the demand to delete a follow record.
> * They don't exactly know how to solve this at this time.

Does Mastodon have this problem, as well?

**Comment by thealexparadox** [comment]
No. There are several ways to remove unwanted followers on Mastodon (soft-blocking, or there's a tool in the user prefs).

**Comment by thealexparadox** [comment]
> Does Mastodon have this problem, as well?

No. There are several ways to remove unwanted followers on Mastodon (soft-blocking, or there's a tool in the user prefs).

**Comment by perigozin** [comment]
> > Even if for some reason implementing this explicitly was something "to be discussed". Is there any technical reason why blocking and unblocking doesn't force an unfollow? If it's just a matter of implementing it, is there any reason why it shouldn't be this way?
> > @markski1
> 
> This is has been answered [here](https://github.com/bluesky-social/social-app/issues/1160#issuecomment-1677642129) but the TL;DR is this:
> 
> * It's impossible to do this since there are users who have servers that are not run by Bluesky.
> * When you follow someone or block someone, it creates a record.
> * The block record basically overrides the behaviour of the follow record, but can't delete the follow record itself.
> * There can (and will) be servers that will simply disrespect or ignore the demand to delete a follow record.
> * They don't exactly know how to solve this at this time.

That's why Bluesky will never be the next big social network. Being so reliant on a broken protocol and refusing to change will be it's demise.

**Comment by IPv6** [comment]
> Being so reliant on a broken protocol and refusing to change will be it's demise.
It`s a normal technical limitation, which actually make protocol robust, with single point of responsibility for user data.
Quite understandable limitation given open nature of all connections, this feature inherently require CENTRALIZATION of PDS control. And with control centralisation a lot worse things may happen down the road (just look into Musk`s X-hole)

May be devs will be able to mitigate the issue by ability to delegate PDS hoster of unwanted follower to do basic stuff? Like unfollow by request from other trusted PDS hosters. That way majority of users will have the feature, and rogue hosters can be just blocked

**Comment by piecevcake** [comment]
> Just out of curiosity: why would you want to remove a follower but not block them?
Because blocking doesn't stop them following you.

**Comment by Everian-H** [comment]
Wanting to chime in here on "why would you want this instead of just blocking" : I'm part of a fairly niche group of users, and we recently put together some starter packs for the community. These starter packs are now being follow en masse by, largely, highly political accounts. I don't need to block these people because they can live their lives, but now I have nearly 100 followers (almost 20% of my total) who have no interest in what I post and only followed me to get a follow back. Feels bad, man.

**Comment by pernsdorff** [comment]
> Hi
> 
> Just out of curiosity: why would you want to remove a follower but not block them?

I don't see a good reason not to be able to manage your followers. There seem to be accounts that automatically follow someone as a way to get themselves followed -- usually do not care about that other account content.
This is a fake a trolling method of gaining metrics for themselves and hoping you click the follow-back.

In my mind the best feature is to allow you to control that a Follow request has to be "accepted".

Blocking and Follow are different unrelated purposeful activities.

**Comment by avazig** [comment]
> I also hate how my followers number is inflated by bots and engagement farmers despite having blocked them.

This was a huge problem on the other place. Even if we disregard the algorithm, having hundreds or thousands of inactive followers did add to the feeling of shouting into the void.

I was to be able to rid myself of low-value followers, and that includes the numbers.

**Comment by TomasHubelbauer** [comment]
I'd also like to curate the list of people that follow me. Blocks serve a different purpose. Removing someone's follow could be an extra record that overrides the behavior of the follow record they published. It is okay for it to be public IMO. If it is not possible to prevent someone from following me due to the decentralized nature of the service, I think hiding your own followers in the list (and the counter on the profile screen) without preventing them from subscribing to your posts would be helpful, too.

**Comment by piecevcake** [comment]
> Hi
> 
> Just out of curiosity: why would you want to remove a follower but not block them?

You can't see posts of accounts you have blocked.
If you unblock them they are still following you.

**Comment by maximedrouhin** [comment]
In my understanding most people here want the ability to remove followers because of spam-follows by bots and people hoping for a follow-back.

Having granular control over your followers is of course a great way to fix this, but maybe this could be accompanied by some form of flagging of abusive follows? With a gradation of warnings and temporary disablement of the follow option, this might be able to prevent most of the problem at the source?

**Comment by hvdijk** [comment]
@TomasHubelbauer hit the nail on the head, IMO: it has not been clearly covered that there are two different reasons for wanting to remove a follower that do not need to be handled the same way:

1. Wanting to remove a follower so that they no longer get notified of your posts
2. Wanting to remove a follower so that others do not see that they are following you

The focus here has been on looking for a single solution that covers both cases. But a single solution does not need to cover both cases:

Case 1 can be handled by voluntary removal. As mentioned, yes, misbehaving servers could fail to honour this. But that is not an issue. This is roughly equivalent to, on other platforms, someone who keeps re-following after a soft block. On those other platforms, there is a simple solution for this: that is reason enough to upgrade the soft block to a hard block.

Case 2 can be handled by having a way to hide followers rather than remove them. There are accounts that follow me where I do not care whether they see what I post, but if someone looks at *my* account to see who is following me, I would rather not have those followers visible for whatever reason. When someone looks at my profile, I should have control over what people see. If some of the data has to come from external servers that BlueSky have no control over, I should still have a say in whether that data is included.

**Comment by Yowlen** [comment]
Honestly, this needs to be fixed sooner rather than later. It's a vulnerability at this point. Transphobes are exploiting this (among other vulnerabilities in the platform) to send thinly veiled threats and transphobic messages to trans people.

They've been making a ton of sockpuppet accounts with these kinds of messages in their bios, mass following trans people and allies, and as they get blocked, they simply make more, rinse and repeat.

The inflation of follower counts might ultimately be a byproduct of this tactic, but when a trans person has more followers that want to kill them than ones that genuinely want to interact with them, it has its own negative effect on our mental health.

**Comment by SystemsOperator** [comment]
Definitely want this and, this request may be related:

https://github.com/bluesky-social/social-app/issues/6330#issuecomment-2521723043

- **Mute YOURSELF (your own posts, others' reposts of you) for another user.**
Unlike blocking, this shouldn't be public. Rather, a safer, private way to lessen chances of interactions from some users.

**Comment by joedinoto** [comment]
Adding my voice to the chorus. Requesting the ability to remove a follower. I joined up to grab my username and immediately had a spam storefront account follow me. I can block it sure, but now my page has "1 follower", even though I have blocked that account. I've also reported the account.

**Comment by davimack** [comment]
There are safety issues not addressed in the general discussion, I think. If someone exhibits creepy behavior and you don't want them to be aware that you've noticed them, for example. They may regard blocking as paying attention to them, create another account, and return. Blocking = encouragement.

**Comment by colinstu12** [comment]
Blocking can cause more alarm and damage than doing nothing. Please add the ability to remove followers. Especially with Bsky making blocks so public too. That or give the ability to block without them knowing somehow. But that will never happen. So yeah, bring the ability to remove a follow.

**Comment by lunaynx** [comment]
I believe a good temporary solution to some of the concerns here that does not require protocol changes can likely just be handled by the AppView, which is to subtract blocked users from the follower and following counts. They already appear to be hidden from the lists, at least for me as the blocker.

**Comment by cardboardmoon** [comment]
I’ve read about 75% of this issue. I’m confused. 

1. Why is a block not an actual block? If I block someone, I expect that person to not be able to see or interact with my account in anyway. You know, like….I’m blocking them. From interacting with my account in any way. Like I’m spawning a literal block of concrete that crushes their entire body. 
2. What are the gymnastics involved with removing a follower? It seems…simple…on paper. I’m a gamedev so I know programming is hard (even, and sometimes especially, the simple-on-paper stuff) but this seems like it should’ve been a launch feature. 

I mean these two questions with no disrespect, just poking fun at the head-scratcher that it is. I love Bluesky and appreciate the massive effort going into it! Best social media experience ever, thank you all!

**Comment by IPv6** [comment]
> I mean these two questions with no disrespect, just poking fun at the head-scratcher that it is

Blocks and followers are simple when there is one database and one company to rule them all. But BlueSky specifically tries to evade centralised solutions. Specifically BlueSky is not in position of sole control over user data, there are limits between how data can be changed inside users local storages. This is like multiplayer games, where games can - and as we live in real world - they definitely would cheat.

So the problem is to have base solution that CAN NOT BE CHEATED. Currently that lead to system with personal blocks lists and removing follower is not that directly possible, since user data handler (game client) can simply reject any removing. This is not enforcable

**Comment by davimack** [comment]
> So the problem is to have base solution that CAN NOT BE CHEATED.

Is there a short-term solution, though?  If that short-term can address 99% of the problem, and doesn't present much of a programming burden, then it would be better than waiting for a long-term solution, particularly as the long-term solution seems intractable within the current architecture.

Cheating happens.  Sometimes a pragmatic approach is better - making it difficult will make it painful to repeatedly violate the block, and so the adversary loses interest.  Sometimes.

**Comment by cardboardmoon** [comment]
> > I mean these two questions with no disrespect, just poking fun at the head-scratcher that it is
> 
> Blocks and followers are simple when there is one database and one company to rule them all. But BlueSky specifically tries to evade centralised solutions. Specifically BlueSky is not in position of sole control over user data, there are limits between how data can be changed inside users local storages. This is like multiplayer games, where gamers can cheat - and as we live in real world - they definitely WILL cheat.
> 
> So the problem is to have base solution that CAN NOT BE CHEATED. Currently that leads to system with list of personal blocks and removing follower is not that directly possible - since user data handler (game client) can simply reject any removing. Removing is not enforceable

I appreciate the explanation!

**Comment by fukahire** [comment]
I know the problem with bluesky's system.
but without this function, bluesky can't be a mature social media.
Even invalid account still stay their follow, this is not make sense.

**Comment by SpriterGors** [comment]
I agree with @fukahire 's sentiment. Perhaps the easy solution for the time being, at least to increase the user's feelings of safety, is to simply hide data instead of actually deleting it. That is:
1. I have 100 followers;
2. User A follows me, and now I have 101 followers;
3. I block User A;
4. User A still stays in my follower list internally - but the programming recognizes it's blocked by me and thus the display system does (totalNumberFollowers - totalBlockedFollowers), that is, 101-1=100;
5. if User B clicks on my follower list, it will recognize User A as blocked and will not show them in the list.

This is not a perfect solution - User A will still know I blocked them, which could generate enmity. However the part of knowing the number of genuine connections I have, and the sensation of being able to curate my own experience, will prevail. But perhaps a quicker temp solution would increase user satisfaction while the gordian knot is being cut.

**Comment by hvdijk** [comment]
> I agree with @fukahire 's sentiment. Perhaps the easy solution for the time being, at least to increase the user's feelings of safety, is to simply hide data instead of actually deleting it.

What you are asking for is already implemented except for the follower count. If Alice follows Bob, Bob blocks Alice, and Charlie looks at Bob's followers, Charlie already does not get to see Alice. I know this because I tested this before posting this comment. What is the reason you think Charlie does still get to see Alice?

This issue is about what happens if Bob subsequently decides to unblock Alice. Your workaround for that is to just keep Alice blocked. For me and for others here, that is not an acceptable workaround.

**Comment by SpriterGors** [comment]
My focus was indeed on the follow count, however I agree, it is not a proper solution in the grand scheme of things. Sorry.

**Comment by ICEknigh7** [comment]
> What you are asking for is already implemented except for the follower count. If Alice follows Bob, Bob blocks Alice, and Charlie looks at Bob's followers, Charlie already does not get to see Alice. I know this because I tested this before posting this comment. What is the reason you think Charlie does still get to see Alice?

Have you checked in Clearsky?

**Comment by hvdijk** [comment]
> Have you checked in Clearsky?

Have you? As far as I can tell, ClearSky would not inform Charlie that Alice follows Bob, regardless of whether Bob blocks or unblocks Alice, because ClearSky never displays followers at all, unless there is some hidden option that I am not seeing.

**Comment by foldsters** [comment]
Here's my take

Each user could have their own public "accepted followers" list, a private "rejected followers" list, a private "pending" followers list, and a private rule setting for whether new followers get added to the pending list or automatically to the accepted followers list automatically (effectively implementing follow requests).

Any user should only be able to be in at most one of these lists, but that will be up to server compliance.

It will be up to the client to decide whether the "followers" number on A's profile will be total followers list (calculated by users following from the outside) or the accepted followers list (maybe the client could call the total followers list "watchers" or similar, since you don't control who watches you).

I think this solves a couple issues, like if a user A follows user B, but A isn't in B's accepted followers list, only B knows whether that user is intensionally rejected or unmeaningfully pending, so it's not the same weight as a block or mute.

This approach doesn't change whether A appears to be following B from A's perspective, but again clients could choose to label A's following list as a "watching" list, and only having users that have A in their accepted followers list be in A's apparent following list.

What do we think of that?

**Comment by Lithdel** [comment]
Kinda crazy how a year and a half later there is still no solution to removing unwanted followers from your follower count number and your Follower list. Literally every single other Social Media Platform has solved this mystery already.

There is no reason why we should have to deal with inflated follower counts due to spam bots following us. If I block someone they should absolutely be removed from my follower list and follower count

**Comment by ICEknigh7** [comment]
How difficult can it be to have the follower counter do:
if(follower is blocked)
 don't count
else
 followers++

**Comment by avazig** [comment]
It's ridiculous now. Since there is no soft block I have to block half a
dozen accounts per day: grifters, porn bots and accounts I want nothing to
do with.

Within a year I'll have thousands of blocked followers inflating my numbers.

It's unacceptable that we don't have ownership of our own followings and
followers.

**Comment by Tamschi** [comment]
> How difficult can it be to have the follower counter do: if(follower is blocked) don't count else followers++

Apparently it might be close (current version): https://github.com/bluesky-social/atproto/blob/d97272de0bef835a8e64e7e924e21e9bb5d012f2/packages/bsky/src/data-plane/server/indexing/plugins/follow.ts#L87-L100

```ts
  const followersCountQb = db
    .insertInto('profile_agg')
    .values({
      did: follow.subjectDid,
      followersCount: db
        .selectFrom('follow')
        .where('follow.subjectDid', '=', follow.subjectDid)
        .select(countAll.as('count')),
    })
    .onConflict((oc) =>
      oc.column('did').doUpdateSet({
        followersCount: excluded(db, 'followersCount'),
      }),
    )
```

That's just for following though, so it would have to be added to https://github.com/bluesky-social/atproto/blob/d97272de0bef835a8e64e7e924e21e9bb5d012f2/packages/bsky/src/data-plane/server/indexing/plugins/block.ts and if you want it to work for list blocks also to https://github.com/bluesky-social/atproto/blob/d97272de0bef835a8e64e7e924e21e9bb5d012f2/packages/bsky/src/data-plane/server/indexing/plugins/list-block.ts (or rather, the calculation should be refactored out to just be reused) and (more problematically) ideally also https://github.com/bluesky-social/atproto/blob/d97272de0bef835a8e64e7e924e21e9bb5d012f2/packages/bsky/src/data-plane/server/indexing/plugins/list-item.ts, though there it would have to fan out to be applied to all who blocked the moderation list and I'm not sure if the current architecture would auto-deduplicate those tasks.

I suspect that last point could get **very** heavy for popular lists that are frequently updated, but other than on list-item the load probably isn't too awful? Definitely much heavier than currently though, unless a DB index that takes blocks into account already is also created. [The current one](https://github.com/bluesky-social/atproto/blob/d97272de0bef835a8e64e7e924e21e9bb5d012f2/packages/bsky/src/data-plane/server/db/migrations/20230309T045948368Z-init.ts#L203-L208) can respond with the raw total follower count immediately, but likely not if you exclude all blocks from that.

**Comment by cooljeanius** [comment]
Looks like the atproto-side issue tracking this is bluesky-social/atproto#1803

**Comment by mschwendener** [comment]
1) You want to prevent people from following you, without blocking them 
2) You don't want to see their profile on your follower list, and they should not be included in your follower count 

For 1), would a function "mute myself for this account" cover that? 

- it would prevent the account from seeing your posts
- it would prevent the account from just following you again after a soft block (= block, then unblock)
   
---

[The second point is discussed in context of blocking in several threads.]

**Comment by weber719** [comment]
This shouldn't be a huge issue, you have my userID, you have the userID of the person I want to unfollow, you use the person's userID to update their database to remove my userID from their follow table. We should have the power to remove followers from our a count. I shouldn't have to block 100s of user accounts because it's the easiest solution for your developers

**Comment by davimack** [comment]
I think that some of the concern is safety.  Over with the bird, you could remove someone from your followers (block/unblock).  This let you disconnect them, but they'd still be able to come back.

_As I understand it_, there is thinking that that they'd be in a sort of grey zone: they could refollow, but hopefully wouldn't.  They wouldn't necessarily feel like they had to get around the block - they weren't blocked.  So, they'd hopefully just go away.  And you could keep removing them until they got the point: that you just don't want to interact with them.

Item 1 would be a bit of an escalation, if you're actively preventing them from following, but I personally wouldn't feel that it was that much of a difference as to worry about it.  Yes, in other words.

I think not seeing your posts is a step beyond, maybe?  I think the desire here is to just not have them interact or participate in your community.

This is me, though, expressing my take on things.

**Comment by musicjunkieg** [comment]
> This shouldn't be a huge issue, you have my userID, you have the userID of the person I want to remove from my followers list, you use the person's userID to update their database to remove my userID from their "following" table. We should have the power to remove followers from our account. I shouldn't have to block 100s of user accounts because it's the easiest solution for your developers 

So you're saying that people should be able to modify your PDS without your permission and how would you even do that if Bluesky isn't hosting the PDS, but the user is self-hosting and has modified the implementation to control deletes? 

I get that you think this should be simple, but a brute force mechanism doesn't make sense in a distributed system.

**Comment by mschwendener** [comment]
> not seeing your posts is a step beyond, maybe?

Thank you – I was just curious.

**Comment by stillhope** [comment]
For me, it is about SAFETY.
Ability to refuse a follow (or unfollow) removes possibility of lurking behaviour.
People deemed 'pesky' I wish to add to the pesky list, to warn others.

I am new to bluesky and have no intention of posting and  my profile has no content either. I love to consume content. 
Already 2 people have followed me. 1 I regard as pesky and other person made sense.

Pesky : person wants me to follow back (like they did in LinkedIn in early days) => I looked at their profile and nothing even remotely related to my interests and not my cuppa of tea.  I have blocked but I don't want them have any connection with me what so ever. Creepy.

Make Sense: I liked a few of their posts via repost of a follower. They can stay.

If it was my child receiving undesired follows, my concern would be greater.
I would actually prefer to set my follow mechanics e.g. allow followers, request-approve, no followers

**Comment by mackuba** [comment]
> Ability to refuse a follow (or unfollow) removes possibility of lurking behaviour.

People can be lurking in many other ways though without technically following, even using RSS for example…

**Comment by stillhope** [comment]
oh. thankyou for the information. Did not realise that either.

**Comment by colinstu12** [comment]
> People can be lurking in many other ways though without technically following, even using RSS for example…

How many people are seriously stalking folks via RSS via lurking on modern social media platforms that have terrible blocking/mute controls? 

Whataboutism stinks and doesn't make it any more right or wrong.

**Comment by va3jme** [comment]
I actually wrote an article on this... on why you need to be able to remove your followers.... hear it is

Note that this article was written January 14, 2025 and represents the authors experience with Bluesky at and up to that time. Things may change following the publication of this article and the author may decide to revisit this article in the future.

I had high hopes for Bluesky, I really did.

But what is Bluesky you say? Well, paraphrased from the [Bluesky Wikipedia article](https://en.wikipedia.org/wiki/Bluesky) it’s a “microblogging social media service”, similar to X (Twitter). It began in 2019 as a research initiative at Twitter, becoming an independent company in 2021. Development for the social app accelerated in 2022 after Elon Musk’s acquisition of Twitter and subsequent severing of ties between the companies. It launched as an invite-only service in February 2023 and opened registrations in February 2024. It reached 20 million users by November 2024 [which is around when I joined.]

And Bluesky is the first social media platform in my opinion that seems to have gotten so many things right, and from the outset! It allows fantastic control over what content you actually see. You can choose among various “feeds” to control what you see and you can even create your own feeds! “The algorithm“ feed, named “Discover” is one of default feeds, and from my experience it does a good job of suggesting content that should match your interests. There’s also “Following” (I can’t recall the exact name) which presents just the content posted by people you follow. You can also select a content feed that is comprised of things that those whom you follow seem to be interested in–it’s kind of a “show me what my friends like.” And you can delete any feed you don’t want, including the default feeds.

And I have tons more great things to say about BlueSky, including its clean, easy-to-understand interface, the availability of a web-based interface, the consistent look-and-feel and usability across platforms. Yes I could go on, but I can’t continue to sing its praises while knowing it has one major flaw, and that flaw is so egregious that it has caused me to leave Bluesky (for now.) That flaw is Bluesky’s lack of follower control. What do I mean by follower control? I mean your ability to prevent people from simply following you, meaning anyone can add you to their follow list, without your consent.

You may be thinking “well so what, why should I care who is following me… have at it!”  Well, how would you feel if one day a white supremacy account decided to follow you on Bluesky. And shortly after that–and based on that account following you–similar hate groups began to follow you. Before long, you have dozens of unwanted white supremacy followers. And racists. And crypto-buds. You will have unwittingly become a marketer of all the things you do not support.

Sure you can simply block them… but here’s the problem. You see, BlueSky uses what is called a federated model for content distribution. You own your data. Sounds great right? Well, that data includes who you choose to follow, which means that the initial white supremacist account who chose to follow you, well, they own their follow data, and so you can’t ask the platform to delete that association because the platform, by design, has chosen not to interfere with peoples data, mainly since it doesn’t own that data!

And worse, even if you do block them, that block only applies to your experience with them. In fact, anyone can see who you have blocked, and who has blocked you just by entering your profile on clearsky.app. They don’t even have to be a member of Bluesky to do this, nor in fact do they need to provide any tracking information at all! So if someone has an agenda, say to spread their word of hate, all they have to do is check the accounts that they support to see who is blocking those accounts. Armed with this information–a list of accounts apparently opposed to their ideology–they can use that list to bombard you with new followers. And before you think that’s difficult, the tools to so are already available, and simple to use! They can simply add you to what is known as a “starter pack” which is a list of accounts packaged up so that anyone can follow all the accounts with just a few clicks. They can create and publish this list, and there’s nothing you can do about it. And so their followers—and yes, bots—can target you (and other like-minded people) specifically. Essentially, the tools are out there to profile you, and target you, for good or for bad.

Still think this is just alarmism? Well, after being a member of BlueSky since November of 2024 (I was a very early adopter), it wasn’t until I started posting regularly in early January 2025 that my followers list started to grow quickly and I soon noticed it contained many accounts whose ideologies I do not support, pretty much as I describe in the scenario earlier. Realizing this I decided to “curate” my followers list which was easy at first, but within just a week I found I was spending more time curating my followers list than I was actually consuming content! This was a problem, and it’s what lead me to investigate it, and subsequently it’s what motivated me to leave, and to write this blog entry.

So as mentioned, I did some investigation on this and found many interesting discussions, but I would like to draw your attention to this one: https://github.com/bluesky-social/social-app/issues/1160#issuecomment-1677642129 

> “This is something we’ve discussed pretty heavily internally — we don’t yet know how to do it reliably.
> 
> The issue is that every user controls their own dataset. When I “follow” somebody, what I’m doing is publishing a follow record on my account. When I block somebody, same thing: I publish a block record. A block record can override the behaviors of a follow record, but it can’t force-remove the follow record from somebody else’s repository
> 
> We have discussed a voluntary removal of the follow record. That is, if somebody blocks me, my server automatically deletes my follow record. The only concern is that it’s impossible to force. A server that’s misbehaving (intentionally or not) wouldn’t delete the follow record, so there would be some confusing scenarios where soft-blocks don’t work. That led to some discussion of whether we could try to enforce voluntary behaviors by social pressure (eg “Follow the protocol or your server will be penalized”) but we haven’t established a framework for that kind of governance yet so we’re hesitant to move forward with it.”

So as you can see, they’re aware of this issue, and that post was from mid August 2023, five months ago and the issue still exists. In fact, Bluesky was opened for general, no-invite-needed availability with this issue still in place, and it’s still an issue as of the time of this writing, nearly two years after invite-only launch.

I’m a software developer, and I can speculate on ways they could fix this, but despite my experience in this area in general, I don’t know the challenges the team face regarding how to fix this issue. More concerning to me is that I have yet to be able to determine their timeline for fixing this issue, including if they intend to fix it at all! And I am not alone in my concern… see github bluesky-social social-app issue [#1160](https://github.com/bluesky-social/social-app/issues/1160) for the ongoing discussion. I have followed this issue so that I can update this blog entry should there (ever) be a resolution.

So here we are. Bluesky… so close, and yet so far! I love everything about it except this one issue, and until it’s resolved, I can’t be a member.

Sigh.

**Comment by IPv6** [comment]
Lets not forget that in exchane of follower remover disability (temporary) BlueSky gives a lot. Gives guarantors no other social network can give. It`s a trade off - and imho it is quite ok. Followers counter is not that important comparing to total lockout of your data for malicious owner, imho

**Comment by ICEknigh7** [comment]
@IPv6 Yeah ok but this is an issue.

**Comment by wojtek-be** [comment]
> For me, it is about SAFETY.
> Ability to refuse a follow (or unfollow) removes possibility of lurking behaviour enable via follow feature.

Maybe you should stop posting online on public social network?

> How many people are seriously stalking folks via RSS via lurking on modern social media platforms that have terrible blocking/mute controls?

Sorry to be harsh but - if everyone wanting this feature are so concerned about safety then… they are doing something wrong.
On the one hand claiming that it's about "safety" and whatnot (which implies that there are hords of people interested in them) and on the other hand dismissing the simple fact that at the end of the day all data is public and easily available. Even if your profile is "only for registered" then what you post can be access not only via RSS but also by simply opening your profile. And if you are claiming that they are "stalking" then it wouldn't be surprise to simply check your profile directly (probably even more so without actually following to not leave any trace…).

Once again - if anyone is concerned about safety then they should reevaluate their online presence and thing twice and trice what they are posting online 🤷 

(one "solution" to this would be to have only list of allowed followers/people that can see the content but without anything short of encrypting each payload with a public key of approved follower I don't see it happening…)

**Comment by iharakazuki** [comment]
Honestly, I think this is one of the most awaited features for Bluesky.

What most users care about is not the protocol standard,

but whether proper communication between users is possible,

and I think deleting followers should be a must for that.

**Comment by zavvy1** [comment]
This is absurd that it wasn't AT LAUNCH. It's what's keeping me using twitter for now. It's been years
 This is inexcusable at this point.

**Comment by ccrowles** [comment]
I actually thought that this was a feature already and was so surprised to notice that it isn't.

I really hope it will be implemented in the future, because people can get very nasty when they notice they've been blocked, vs when you can remove the follow it's perceived not as harsh.
I removed a few really creepy people including a stalk-ish ex from my twitter followers in the past, and I would really appreciate if a similar feature would be implemented in bluesky. I'd feel so much safer.

**Comment by danielah05** [comment]
this entire thing gets worse when u realize that suspended accounts dont get removed from ur followers either.
right now im getting kinda mass followed by bots (looks like theres a wave going on) and its boosting my follower count, some of them are getting suspended but my follower count is not going down.
this sounds INCREDIBLY abusable and really needs to get fixed, if u ever get targeted by a mass bot follow wave (i have seen this happen a lot a few years ago on twitter), ur follower count will forever be stained from it.
genuinely there being 0 way for bluesky to remove people from ur followers besides the person themselves unfollowing is really bad, this is a massive problem.

**Comment by IPv6** [comment]
Suggestion to developers: may be add "public followers" and "broad followers" counts? 
- "broad followers" are the same as for now, fast and less-obvious way to one-way addition.
- "public followers" can be GENERATED via some "vacuum" or "cleanup" or "followers update" procedure, initiated by user (not faster than once a day). Where user PDS checks each follower and removes suspended/blocked/muted and optionally non-mutuals. This procedure could use some protocol-specific negotiated crypto-handshake with other PDS to request self-remove from different PDS and in case of third-party PDS is not behave correctly - user can be given precise feedback who is behave wrongly here. Such problematic PDS can be labelled as such for moderation automatically since they clearly perform something scammy (bots etc)

**Comment by Bmaster4114** [comment]
I'm adding my 2 cents in since, I'm still kinda shocked it hasn't been implemented yet.

There are a few points I want to make, I get that the system tracks it multiple places, if I follow someone it shows in my list I follow them, and on their list someone (me) just followed) if I unfollow, it removes me from their list, and them from my respective lists. 

If I block them after following, I am still on their follower list (but hidden). If they block me I am still on their follower list. If I follow a block list that auto blocks them I still follow them, and again vice versa if they sub to a list I am on. In the case of a personal I click the block button it should act as an automatic unfollow, if I or they deactivate, or have the account suspended it SHOULD act as an unfollow. Now I can agree that block lists are a slippery slope on if they should or shouldn't but if either account actively clicks block on the other, or if either account no longer exists, then they shouldn't be in EITHER person's follower/following list.

For me its a case of the "follower count" and bot scams and such, nothing is stopping a bot/scammer to mass follow people get followed back and then BLOCK those people locking in the follower count. People see the large follower count and think "Wow big number! must be legit. There would be no way currently to prevent this. 
Meanwhile I do think its bad that this problem ISN'T being addressed or hasn't been stated, because if/when it IS changed, people are going to cry out that their accounts got "follower purged" if they are blocking scam followers/ blocked by people that followed but didn't unfollow before blocking. Users **DO NOT KNOW** how many people actually follow them, and the only way to know for sure is to load the full list and either manually count or use F3/ word search type in "follows you" and see the ACTUAL NUMBER (barring any false positives in user's bios, or other cases).

Currently my one account says I have 1070 followers, I actually only have 977 show up on the list. that is **93** accounts I either block, or blocked me before unfollowing, OR got suspended/deactivated.

Others have mentioned specific reasons for why it would be a good feature to have for personal safety, but its also a general case of being clear and honest. Showing the true follower counts (and all around removing follows) would help prevent scammers from taking advantage of the system to seem legitimate. While also allowing folks to see how many people are ACTUALLY following them verses how much engagement they get on posts. This is something that I am still shocked hasn't been updated, or at the very least seems to be unmentioned ANYWHERE. I've seen some articles online erroneously saying "Blocking removes follows" because that's the NORM. I hope this gets changed, BUT I also hope IF AND WHEN IT GETS CHANGED that it is HEAVILY explained so people don't panic about their follower counts dropping.

**Comment by i7smoon** [comment]
> This is something we've discussed pretty heavily internally -- we don't yet know how to do it reliably.
> 
> The issue is that every user controls their own dataset. When I "follow" somebody, what I'm doing is publishing a follow record on my account. When I block somebody, same thing: I publish a block record. A block record can override the _behaviors_ of a follow record, but it can't force-remove the follow record from somebody else's repository.
> 
> We have discussed a voluntary removal of the follow record. That is, if somebody blocks me, my server automatically deletes my follow record. The only concern is that it's impossible to force. A server that's misbehaving (intentionally or not) wouldn't delete the follow record, so there would be some confusing scenarios where soft-blocks don't work. That led to some discussion of whether we could try to enforce voluntary behaviors by social pressure (eg "Follow the protocol or your server will be penalized") but we haven't established a framework for that kind of governance yet so we're hesitant to move forward with it.
> 
> A subsequent proposal to require "follow requests" has been discussed ([bluesky-social/atproto#1352](https://github.com/bluesky-social/atproto/issues/1352)) which is technically feasible but adds a little more friction than we want for the default model.
> 
> That leaves us where we currently are, which is tracking these two options but not ready to move on either yet.

I'm not familiar at all with how Bluesky works internally so ignore me if this doesn't make sense, but instead of needing to remove someone else's follow record, why shouldn't I be able to publish on my own dataset a record that explicitly cancels any past follows from another user? Instead of removing, it would be more like overlaying.

**Comment by mackuba** [comment]
@itsmoonx3 yeah, that's what I was suggesting before too:

> Copying here what I wrote on Bluesky about how this could be handled:
> 
> We could have some additional record type for this purpose, let's say an "unfollow" or "softblock" or "denyFollow" record. This would be like a reverse, negative follow, a record that you'd add to your repo, pointing in the opposite direction, indicating that you don't want to be followed by that account (it would be a permanent effect until you delete the record, not a one-time thing like classic "soft blocking" is on Twitter; the user would see the follow button disabled on your profile).
> 
> The way it would work is that AppView would take that into account and would make it seem as if the user is not following you, even though they technically have a follow record pointing to you. It would effectively make that follow from them to you disappear, even though the record is there.
> 
> The precedent for this is how posts are handled in AppView in the context of blocks and thread gates. If someone replies to you and you block them, that comment "disappears", even though there is still a post record in the author's repo - but AppView pretends that it doesn't exist. In the same way, it would pretend that the follow record doesn't exist.

I think the possible issues with this are:

- should it be one-time or permanent?
- is it a problem that it needs to be public?

**Comment by i7smoon** [comment]
> I think the possible issues with this are:
> 
> * should it be one-time or permanent?
> * is it a problem that it needs to be public?

@mackuba I personally think it should work like a negative / reverse follow record like you said, however I don't think it should be permanent in the way you described.

I think it should **disable any _past_ follow records** from another user.

It should be **irreversible**: once you disable (remove) the follower you do not have the ability to regain that follower yourself.

But **not permanent**: the other user should always be able to follow you again as long as you haven't blocked them, in which scenario you should be able to disable their follow again.

I think blocking another user should also automatically issue this reverse follow record, automatically removing them from your followers.

I would also try to avoid including any front-end indicators that this reverse follow occurred. Ideally, it should appear to all accounts the same way it would if the other user had never followed you.

But like I said, I am not familiar on how Bluesky works internally, so I am not sure if this is technically possible to implement.

**Comment by davimack** [comment]
May I throw in an analogy?

Let's think of this as being like one of those sushi boat restaurants, where the items continually scroll past the diner.  The diner puts up a sign for the sushi chef, "More Edamame!"  And the chef puts out more edamame, which scrolls past the diner, who may interact with it.  The chef may override that order, though, and say, "No more edamame, he eats it all!," putting up a "no edamame" sign in front of that diner.  And suddenly there's no more edamame at all for that diner.  The chef only has that one option: block.

We've been talking about are things like letting the chef put down the sign, or take away the sign without putting up the "no edamame" sign.

1. Can they see content in the stream?
2. Can they see content when they visit the chef directly?
3. Does it look like they have a sign up asking for content?
4. Does the chef still get notified if they interact with content?  (i.e., they come to the account and directly comment on content)

Some of these feel almost related to shadowbanning, which should probably be considered alongside of these.

**Comment by Tamschi** [comment]
> It should be **irreversible**: once you disable (remove) the follower you do not have the ability to regain that follower yourself.

This part here isn't compatible with eventual consistency in ATProto since you could remove the record that cancels previous follows at any time and later-initialising app servers wouldn't easily see it.

The other parts can be eventually-consistent between scraping app servers if you make use of (self-reported) timestamps to figure out which follow records are affected by the remove-follower record. Problems could arise if a (misbehaving?) PDS reported a future timestamp for a follow that should be cancelled, so that should ideally be taken into account when creating the cancelling record.

**Comment by Solenhal** [comment]
Giving users the agency to self-moderate who follows them would be awesome.

**Comment by ReimiMJ** [comment]
Hi 

Why would you want to remove a follower but not block them? I wan't block my old death followers, thanks.

**Comment by MarkROR** [comment]
**_Someone new followed you..._**

<img width="610" height="351" alt="Image" src="https://github.com/user-attachments/assets/9cee728a-9433-4474-b1b7-ae4f1312b900" />

And they will **always** appear on your followers list, **for everyone to see, forever.** 
Free scam decoration, enjoy. Especially noticeable if you don't have 300+ followers. Enjoy 👍

Yeah, no, totally not a problem, unnecessary feature, can see why we're still debating this after 3+ years.

**Comment by davimack** [comment]
Blocking is not neutral: it’s a reaction. For trolls or stalkers, being blocked is often a win because it confirms impact. Quietly removing a follower denies that dopamine hit and avoids feeding the behavior.

**Comment by colinstu12** [comment]
Safe spaces demand the ability to remove troubling individuals. To suggest otherwise is… only encouraging to the attackers. 

Not to mention the fact that blocks don't decrement the Follower count also artificially inflates the numbers game. Folk's numbers should not be inflated with fake bot accounts.

**Comment by surfdude29** [comment]
> **_Someone new followed you..._**
> 
> <img alt="Image" width="610" height="351" src="https://private-user-images.githubusercontent.com/63402074/547249353-9cee728a-9433-4474-b1b7-ae4f1312b900.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzA2Njg1MjcsIm5iZiI6MTc3MDY2ODIyNywicGF0aCI6Ii82MzQwMjA3NC81NDcyNDkzNTMtOWNlZTcyOGEtOTQzMy00NDc0LWIxYjctYWU0ZjEzMTJiOTAwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAyMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMjA5VDIwMTcwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTgyODE4YTM4MzRhMzdhNzA2NmEwYTQ1NjcxYWY5NGY1Zjc5MTQ5MjExY2FhNDc3ZGNlYzgwY2I3MTE3NmU1OTImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.sUzOdXiU8Xn70sxzVfGMlJFOswonj9OnrbAXsSPdiuY">
> And they will **always** appear on your followers list, **for everyone to see, forever.** Free scam decoration, enjoy. Especially noticeable if you don't have 300+ followers. Enjoy 👍
> 
> Yeah, no, totally not a problem, unnecessary feature, can see why we're still debating this after 3+ years.

If you block an account, it will no longer be visible to other people in your list of followers.

**Comment by colinstu12** [comment]
> it will no longer be visible to other people in your list of followers.

no longer visible, but the Follower Count does not update to reflect that fact. (currently)

**Comment by surfdude29** [comment]
> > it will no longer be visible to other people in your list of followers.
> 
> no longer visible, but the Follower Count does not update to reflect that fact. (currently)

Correct

**Comment by Bmaster4114** [comment]
> I'm adding my 2 cents in since, I'm still kinda shocked it hasn't been implemented yet.
> 
> There are a few points I want to make, I get that the system tracks it multiple places, if I follow someone it shows in my list I follow them, and on their list someone (me) just followed) if I unfollow, it removes me from their list, and them from my respective lists.
> 
> If I block them after following, I am still on their follower list (but hidden). If they block me I am still on their follower list. If I follow a block list that auto blocks them I still follow them, and again vice versa if they sub to a list I am on. In the case of a personal I click the block button it should act as an automatic unfollow, if I or they deactivate, or have the account suspended it SHOULD act as an unfollow. Now I can agree that block lists are a slippery slope on if they should or shouldn't but if either account actively clicks block on the other, or if either account no longer exists, then they shouldn't be in EITHER person's follower/following list.
> 
> For me its a case of the "follower count" and bot scams and such, nothing is stopping a bot/scammer to mass follow people get followed back and then BLOCK those people locking in the follower count. People see the large follower count and think "Wow big number! must be legit. There would be no way currently to prevent this. Meanwhile I do think its bad that this problem ISN'T being addressed or hasn't been stated, because if/when it IS changed, people are going to cry out that their accounts got "follower purged" if they are blocking scam followers/ blocked by people that followed but didn't unfollow before blocking. Users **DO NOT KNOW** how many people actually follow them, and the only way to know for sure is to load the full list and either manually count or use F3/ word search type in "follows you" and see the ACTUAL NUMBER (barring any false positives in user's bios, or other cases).
> 
> Currently my one account says I have 1070 followers, I actually only have 977 show up on the list. that is **93** accounts I either block, or blocked me before unfollowing, OR got suspended/deactivated.
> 
> Others have mentioned specific reasons for why it would be a good feature to have for personal safety, but its also a general case of being clear and honest. Showing the true follower counts (and all around removing follows) would help prevent scammers from taking advantage of the system to seem legitimate. While also allowing folks to see how many people are ACTUALLY following them verses how much engagement they get on posts. This is something that I am still shocked hasn't been updated, or at the very least seems to be unmentioned ANYWHERE. I've seen some articles online erroneously saying "Blocking removes follows" because that's the NORM. I hope this gets changed, BUT I also hope IF AND WHEN IT GETS CHANGED that it is HEAVILY explained so people don't panic about their follower counts dropping.

I just want to add an addition 2 cents and maybe this is more of a question. 

There is a 3rd party tool called "Cleanfollow" that allows users to link to their bluesky account to, to remove accounts YOU followed if the account you followed was suspended or deleted. This is the ONLY way to remove that follow. I don't know if that is still the case, but the concept that someone's account being deleted or removed not automatically removing the followers of that account doesn't make sense in my mind. The number is artificially inflated on everyone's personal pages. Now if you have over 100 accounts you follow sure you wouldn't notice if there is one or 2 not listed, but if you had 5 accounts followed, it shows 5 on your page, but only 4 are listed and NO one blocks you and you didn't block them (no list blocking either) that follower digit is counted for an account that does not exist. 

Going into that further IF this is still the case, does that mean an account that follows you doesn't get removed if they get deleted or suspended, its the same issue of accounts not being removed from follower/following when blocked, but for accounts that no longer exist? 

The following and follower system I would think would be the simplest functionality, a 3rd party tool shouldn't be needed to do something that is functionally possible otherwise, especially for something as arbitrary as "accounts you follow who deleted their page or got suspended", but its function. 

I could harp and repeat everything I said almost a year ago at this point, but it will be a problem IF this eventually gets fixed and users suddenly see a drop in their follower accounts and assume there was an account purge, OR if engagement/ advertisement/ revenue systems are introduced and accounts try to market and grow on Bluesky, have 50k followers, but only 10k are shown in the list (due to blocking or account deletion). If only those 10k real followers interact with the content will the possible business algorithm consider the REAL follower count 10k, or the fake follower count 50k. It could kill all engagement for users leading to them turning to other platforms. While yes this is a hypothetical, but its a problem that could be prevented if the follower/following system was simply fixed or functioned like almost every other platform.



week later I want to tack on another 2 cents. 

One of the main reasons someone would want to remove a follower from their follower list is because they know/assume a block would raise a stink, a block is noticeable, you can't see their stuff anymore. With clearsky especially you can see EVERY account a user blocks or is blocked by. An unfollow doesn't leave a mark (on most sites) it could be a 'bug', or maybe the person just "didn't actually click the button'. As an example if someone follows you and you block them, they could raise a stink and post false info or lies and generally attack the person out of vengeance, they could circumvent the block and make a new account and harass the person, an unfollow, they don't know their follow was removed unless they saw it happen. Ofcourse everything I said would be bannable if someone block evades or harasses someone they could get their account suspended, BUT how the blueskyteam may not be quick enough to suspend the problem account, and the issue wouldn't happen if they user was just quietly silently marked as removed from the follower list.

**Comment by axelpina** [comment]
The lack of this feature has single-handedly made me not use bluesky at all, while I understand the technicality behind it, it feels like a complete oversight from the team. Now we have to live with hundreds of dead accounts following us which is not indicative of the people we interact with. 

For example I have 400+ followers and only interacted with around 5-15 people. If a user wants to reset their profile they have to create a completely new different profile. Incredibly inefficient, and again, an oversight from the team.

**Comment by jakedane** [comment]
I'm new to Bluesky. I follow a few FOSS & EU accounts. I haven't posted anything. Yet several accounts follow me. Why? Not people I know, not related to the topics I mentioned, and again I haven't posted anything. All the followers on my accounts are bots.

This one https://bsky.app/profile/spkmmt.bsky.social follows 7.4k people. Or this one https://bsky.app/profile/michellehogan.bsky.social follows 397.6k (!!) people and targets "the elderly". No real person follows that many accounts. I've already reported the accounts but apparently operating spam / scam bots is allowed on Bluesky. I've blocked them but they still count as followers on my profile.

👎

This needs an option to approve followers or to be able to remove them, as on Mastodon.

**Comment by axelpina** [comment]
> The lack of this feature has single-handedly made me not use bluesky at all, while I understand the technicality behind it, it feels like a complete oversight from the team. Now we have to live with hundreds of dead accounts following us which is not indicative of the people we interact with.
> 
> For example I have 400+ followers and only interacted with around 5-15 people. If a user wants to reset their profile they have to create a completely new different profile. Incredibly inefficient, and again, an oversight from the team.

After posting this, I erased my account to clean my followers and as as soon as I erased it, my handle got snatched by a bot. My handle was a unique name under the .bsky.social. Couldn't snatch it on time because obviously the bot was faster than me. 

Awful system, won't be seeing me in this platform anymore.

**Comment by Bmaster4114** [comment]
> There is a 3rd party tool called "Cleanfollow" that allows users to link to their bluesky account to, to remove accounts YOU followed if the account you followed was suspended or deleted. This is the ONLY way to remove that follow. I don't know if that is still the case, but the concept that someone's account being deleted or removed not automatically removing the followers of that account doesn't make sense in my mind. The number is artificially inflated on everyone's personal pages. Now if you have over 100 accounts you follow sure you wouldn't notice if there is one or 2 not listed, but if you had 5 accounts followed, it shows 5 on your page, but only 4 are listed and NO one blocks you and you didn't block them (no list blocking either) that follower digit is counted for an account that does not exist.

Guess I'm adding more 2 cents, this may belong in another thread, but I just noticed an issue with the follower count in general. I have 2 accounts one with less than 100 followers, and one with more than 1000 followers. I noticed randomly that the follower count was 1 user less when viewed by my smaller account rather then on my main account. I asked a friend what they saw they saw the count included the 1 user. 

I then checked the differences on their account with my 2 accounts and the opposite was true, my small account saw 1 additional person, while my big account saw 1 less. I checked another person's account, the same was true 1 more when viewed on small.  I checked a bigger third account the difference was THREE users. This was the case on both mobile and desktop, and even when my big account gained followers the one count difference was still there. NOW what I also noticed I couldn't find this discrepancy for accounts that had less than 1000 followers for the ones I checked. The reason I bring this up at all is because now I'm not sure how Bluesky shows the count of followers. I assumed they counted every user whether deleted, blocked, or suspended, but it seems like randomly a different user may not see that same number, so is the site actually getting the hard number or an estimated number? How is the number different when viewed by different people, if it should be getting the same information on both? Is it actually excluding some accounts from the follower count, or is it just a math rounding error?

**Comment by gpbl** [comment]
Yeah, I just created an account for a personal project and was immediately followed by two "adult" profiles. I can't remove them, and they're currently my only two followers 😞
